x-pack/legacy/plugins/maps/public/layers/styles/vector/style_meta.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  IFieldMetaDescriptor,
+  IStyleMetaDescriptor,
+  IGeometryTypesDescriptor,
+} from '../../../../common/descriptor_types';
+
+export class StyleMeta {
+  private _descriptor: IStyleMetaDescriptor;
+  constructor(styleMetaDescriptor: IStyleMetaDescriptor) {
+    this._descriptor = styleMetaDescriptor;
+  }
+
+  _getGeometryTypes(): IGeometryTypesDescriptor {
+    return this._descriptor.geometryTypes
+      ? this._descriptor.geometryTypes
+      : {
+          isPointsOnly: false,
+          isLinesOnly: false,
+          isPolygonsOnly: false,
+        };
+  }
+
+  getFieldMetaDescriptor(field: string): IFieldMetaDescriptor | unknown {
```

Comment:
Not sure `IFieldMetaDescriptor` is needed. Why do you have an interface to describe disparate object types? Should just return `IRangeFieldMeta | ICategoryFieldMeta`

Reply:
parameter should be `fieldName` if its a string. `field` has been used for an object. fieldName is used for the string

Reply:
decided here to remove abstraction and just have separate getters for range or categorical field meta. this removes the need for clients to downcast when retrieving the field-meta.